rust/connlib/tunnel/src/control_protocol/client.rs

```diff
@@ -224,11 +228,27 @@ where
             .get(&gateway_id)
             .ok_or(Error::UnknownResource)?
             .clone();
-        peer_connection.set_remote_description(rtc_sdp).await?;
 
         self.role_state
             .lock()
             .activate_ice_candidate_receiver(gateway_id, gateway_public_key);
+        let tunnel = self.clone();
+        // RTCIceTransport::start blocks until there's an ice connection.
+        tokio::spawn(async move {
+            if let Err(e) = peer_connection
+                .start(&rtc_ice_params, Some(RTCIceRole::Controlling))
```

Comment:
Can we move this into `new_tunnel`?

Reply:
Together with the other suggestion of making `new_tunnel` not `async`, this would avoid this `tokio::spawn` :)

Reply:
Any state modification has to be done after `start` has run, since if successful it means we have a connection. So, we don't want to move it into `new_tunnel` (unless we want to make it `async` again.

Reply:
Hmm, I'd like to get rid of the `self.clone` because for as long as we have `Arc<Tunnel>`, we will have to keep the locks around :(

Reply:
We don't have to solve this in here though, I'll have a play once this is merged!